src/main/java/com/flansmod/common/guns/GunType.java

```diff
@@ -1740,4 +1776,151 @@ public float GetRecommendedScale() {
+        //how much the aim goes back down after shooting. lower is better. 0-2, default 1f
+        public float fall;
+        //how much spraying increases recoil over time. lower is better
+        public float increase;
+        //modifies how much sneaking modifies recovery (usually 10%). default 1%, lower is better, 0-2
+        public float sneak;
+        //how much walking speed affects recoil. lower is bettr
+        public float speed;
+
+        public float sprayLength;
+        public float antiRecoil;
+
+        public GunRecoil(float vertical, float horizontal, float recovery, float recoveryScope,
+                         float fall, float increase, float sneak, float speed) {
+            this.vertical = vertical;
+            this.horizontal = horizontal;
+            this.recovery = recovery;
+            this.recoveryScope = recoveryScope;
+            this.fall = fall;
+            this.increase = increase;
+            this.sneak = sneak;
+            this.speed = speed;
+        }
+
+        //create empty recoil
+        public GunRecoil() {
+            this(0, 0, 0, 0, 0, 0, 0, 0);
+        }
+
+        //copy
+        public GunRecoil(GunRecoil gunRecoil) {
+            this(gunRecoil.vertical, gunRecoil.horizontal, gunRecoil.recovery, gunRecoil.recoveryScope,
+                    gunRecoil.fall, gunRecoil.increase, gunRecoil.sneak, gunRecoil.speed);
+        }
+
+        public GunRecoil read(String[] split) {
+            vertical = read(split, 1, vertical);
+            horizontal = read(split, 2, horizontal);
+            recovery = read(split, 3, recovery);
+            recoveryScope = read(split, 4, recoveryScope);
+            fall = read(split, 5, fall);
+            increase = read(split, 6, increase);
+            sneak = read(split, 7, sneak);
+            speed = read(split, 8, speed);
+
+            if (split.length < 2) {
+                horizontal = vertical * 0.3f;
+            }
+            return this;
+        }
+
+        private float read(String[] split, int i, float alt) {
+            return split.length > i ? Float.parseFloat(
+                    split[i].indexOf('=') == -1 ? split[i] : split[i].substring(split[i].indexOf('=') + 1))
+                    : alt;
+        }
+
+        public GunRecoil copy() {
+            //
+            //  if (horizontal == 2 && recovery == 1 && recoveryScope == 1 && sneak == 1 && speed == 1) {
+
+            String rc = "Recoil 1.9 horizontal=1.9 recovery=1 recoveryScope=1 fall=0.9 increase=2 sneak=1 speed=1.2";
+            //   if(true)
+            //     return new GunRecoil().read(rc.split(" "));
+            //     {}
+
+            //     }
+
+            return new GunRecoil(this);
+        }
+
+        public void applyModifier(float recoilMultiplier) {
+            vertical *= recoilMultiplier;
+            horizontal *= recoilMultiplier;
+        }
+
+        public void addRecoil(GunRecoil recoil) {
+            Random rand = new Random();
```

Comment:
This needs to be changed to a static random that can be reused, for efficiency and neatness